framework/tlvf/tlvf.py

```diff
@@ -851,10 +851,11 @@ def addClassParam(self, obj_meta, param_name, param_type, param_type_info, param
             if (param_val_const or param_val):
                 if is_int_len or is_const_len:
                     if param_type_info.type != TypeInfo.STD: self.abort("%s.yaml --> only std types are allowed in val / val_const" % self.yaml_fname)
-                    lines_cpp += ["for (size_t i = 0; i < %s; i++){" % (str(param_length)),
-                                 "%sm_%s[i] = %s;"% (self.getIndentation(1), param_name, str(param_val) if param_val != None else str(param_val_const)),
-                                 "}",
-                                 ""]
+                    lines_cpp += ["if (!m_parse__) {",
```

Comment:
It would be clearer both for the generated code and the code generator itself if the parse condition was there only once. So for tlvf, that means to swap the conditions `if (param_val_const or param_val)` and `if is_int_len or is_const_len` and then to merge the two int/const len conditions that follow each other, and then to move the generated `if (!m_parse__)` to the beginning of that condition. Am I making sense?

(nitpick).